src/bootstrap/test.rs

```diff
@@ -802,53 +804,52 @@ impl Step for RustdocGUI {
     }
 
     fn run(self, builder: &Builder<'_>) {
-        if let (Some(nodejs), Some(npm)) = (&builder.config.nodejs, &builder.config.npm) {
-            builder.ensure(compile::Std { compiler: self.compiler, target: self.target });
-
-            // The goal here is to check if the necessary packages are installed, and if not, we
-            // display a warning and move on.
-            let mut command = Command::new(&npm);
-            command.arg("list").arg("--depth=0");
-            let lines = command
-                .output()
-                .map(|output| String::from_utf8_lossy(&output.stdout).to_string())
-                .unwrap_or(String::new());
-            if !lines.contains(&" browser-ui-test@") {
-                println!(
-                    "warning: rustdoc-gui test suite cannot be run because npm `browser-ui-test` \
-                     dependency is missing",
-                );
-                println!(
-                    "If you want to install the `{0}` dependency, run `npm install {0}`",
-                    "browser-ui-test",
-                );
-                return;
-            }
+        let nodejs = builder.config.nodejs.as_ref().expect("nodejs should be present");
+        let npm = builder.config.npm.as_ref().expect("npm should be present");
 
-            let out_dir = builder.test_out(self.target).join("rustdoc-gui");
-            let mut command = builder.rustdoc_cmd(self.compiler);
-            command.arg("src/test/rustdoc-gui/lib.rs").arg("-o").arg(&out_dir);
-            builder.run(&mut command);
+        builder.ensure(compile::Std { compiler: self.compiler, target: self.target });
 
-            for file in fs::read_dir("src/test/rustdoc-gui").unwrap() {
-                let file = file.unwrap();
-                let file_path = file.path();
-                let file_name = file.file_name();
+        // The goal here is to check if the necessary packages are installed, and if not, we
+        // display a warning and move on.
+        let mut command = Command::new(&npm);
+        command.arg("list").arg("--depth=0");
+        let lines = command
+            .output()
+            .map(|output| String::from_utf8_lossy(&output.stdout).to_string())
+            .unwrap_or(String::new());
+        if !lines.contains(&" browser-ui-test@") {
+            println!(
+                "warning: rustdoc-gui test suite cannot be run because npm `browser-ui-test` \
+                 dependency is missing",
+            );
+            println!(
```

Comment:
This should print to stderr I think. Also "warning" is no longer accurate.
```suggestion
            eprintln!(
                "error: rustdoc-gui test suite cannot be run because npm `browser-ui-test` \
                 dependency is missing",
            );
            eprintln!(
```